frontend/src/components/files/ExtendedImage.vue

```diff
@@ -79,12 +82,68 @@ export default {
   },
   watch: {
     src: function () {
-      this.scale = 1
-      this.setZoom()
       this.setCenter()
     }
   },
   methods: {
+    fit() {
+      let img = this.$refs.imgex
+
+      const wScale = window.innerWidth / img.clientWidth
+      const hScale = window.innerHeight / img.clientHeight
+
+      this.scale = Math.min(wScale, hScale)
+      this.minScale = this.scale
+      this.setZoom()
+    },
+    refit() {
+      const target = this.fitScreenTarget()
+      this.doMove(target[0], target[1])
+    },
+    fitScreenTarget() {
+      if (this.scale <= this.minScale) {
+        let style = this.$refs.imgex.style
+        let posX = this.pxStringToNumber(style.left)
+        let posY = this.pxStringToNumber(style.top)
+        return [this.position.center.x - posX, this.position.center.y - posY]
+      }
+      else {
```

Comment:
else is redundant since you return in upper if

Reply:
When image is zoomed, else comes into play.